
From: Geert Uytterhoeven <geert@linux-m68k.org>

Amiga frame buffer: We used to have a local hack in fbmem.c to always call the
fbdev setup() routines, even when an fbdev was explicitly disabled on the
kernel command line (video=xxx:off).  This allowed amifb to suspend the
monitor, but program the sync generator of the video controller in Denise/Lisa
to a 31 kHz/70 Hz mode, increasing the maximum audio playback frequency. 
Thanks to the recently introduced fb_get_options() routine, we can kill the
local hack and just use the return value of fb_get_options().

Signed-off-by: Geert Uytterhoeven <geert@linux-m68k.org>
Signed-off-by: Andrew Morton <akpm@osdl.org>
---

 25-akpm/drivers/video/amifb.c |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff -puN drivers/video/amifb.c~amifb-use-new-amifboff-logic-to-enhance-audio-experience drivers/video/amifb.c
--- 25/drivers/video/amifb.c~amifb-use-new-amifboff-logic-to-enhance-audio-experience	2004-10-02 18:41:08.945444528 -0700
+++ 25-akpm/drivers/video/amifb.c	2004-10-02 18:41:08.951443616 -0700
@@ -1239,8 +1239,6 @@ int __init amifb_setup(char *options)
 		if (!strcmp(this_opt, "inverse")) {
 			amifb_inverse = 1;
 			fb_invert_cmaps();
-		} else if (!strcmp(this_opt, "off")) {
-			amifb_video_off();
 		} else if (!strcmp(this_opt, "ilbm"))
 			amifb_ilbm = 1;
 		else if (!strncmp(this_opt, "monitorcap:", 11))
@@ -2260,8 +2258,10 @@ int __init amifb_init(void)
 #ifndef MODULE
 	char *option = NULL;
 
-	if (fb_get_options("amifb", &option))
+	if (fb_get_options("amifb", &option)) {
+		amifb_video_off();
 		return -ENODEV;
+	}
 	amifb_setup(option);
 #endif
 	if (!MACH_IS_AMIGA || !AMIGAHW_PRESENT(AMI_VIDEO))
_
